assignment/impl/src/java/org/sakaiproject/assignment/impl/conversion/AssignmentConversionServiceImpl.java

```diff
@@ -78,7 +79,10 @@ public String deserialize(JsonParser p, DeserializationContext ctxt) throws IOEx
             }
         });
 
-        xmlMapper = new XmlMapper();
+        XMLInputFactory xmlInputFactory = XMLInputFactory.newFactory();
+        xmlInputFactory.setProperty(WstxInputProperties.P_MAX_ATTRIBUTES_PER_ELEMENT, 8000);
```

Comment:
Are these still able to be changed without a code change or do these need to be in serverConfigurationService?

Reply:
I wasn't successful in setting them via java system properties or in sakai.properties, and I couldn't find anything in the woodstox docs that suggested that you should be able to do that.

So AFAIK they must be set in code. An upper limit for the second property (max attribute size) could be found for a specific database by:

    select MAX(LENGTH(XML)) FROM ASSIGNMENT_SUBMISSION;

but I'm not aware of any way to figure out the required max attributes per element in advance.

Reply:
I've made that query in my database and I've found one submission with 6800628 length in the XML, the second one is under 4500000, so it would be great to be able to change this values easily. Maybe you can use a job parameter like the "Configurable Job Test" here:
https://github.com/sakaiproject/sakai/blob/master/jobscheduler/scheduler-test-component/src/webapp/WEB-INF/components.xml#L8-L45
In this case you can set some values before run the job to adjust to your database size.

Reply:
Thanks @juanjmerono i like that idea!!!!

Reply:
You could just make the code run the query itself at the start of the conversion and set the max size accordingly, which would be one less thing for people to configure.

The max attributes per element would still need to be configurable (unless I've missed some mysql XML expression trick to get mysql to figure that out).